### ISSUE TITLE
LLVM assertion error

### ISSUE THREAD

**Issue by loxxous** [ispc/ispc#1633]
Here is some code which has valid ISPC syntax but causes LLVM to die.
```C++
void MyFunc(uniform int x) {
	print("Hello %\n", x);
}

struct MyObject {
	void (* MyFuncPtr)(uniform int);
};

export void TestMain()
{
	MyObject obj;
	obj.MyFuncPtr = MyFunc;

	uniform int x = 60;
	obj.MyFuncPtr(x);
}

export void TestMainBroken()
{
	uniform MyObject obj; // The only difference
	obj.MyFuncPtr = MyFunc; 

	uniform int x = 60;
	obj.MyFuncPtr(x);
}
```

**Comment by dbabokin** [comment]
What command line are your using?
I assume you are using 1.12.0, right?
I've tried this test on Compiler Explorer and it compiles fine https://godbolt.org/z/km9L2g

**Comment by loxxous** [comment]
Yes I am using 1.12.0, the output is from VS 2019. I have also seen that it compiles in Compiler Explorer but not VS. I can package up the solution for you on Monday so you can reproduce the results.

**Comment by dbabokin** [comment]
It passes for me on Windows as well. So I guess the key is in the command line parameters that you are passing to ispc. So we need the command line that you are using (or VS solution).

**Comment by loxxous** [comment]
Here's the solution, just open in VS2019 and you should see the assertion when attempting to build.
[project.zip](https://github.com/ispc/ispc/files/4085697/project.zip)
Hope this helps.

**Comment by aneshlya** [comment]
Thanks!
It is reproduced on Windows/Linux with `--target=avx2-i64x4,sse4`

**Comment by nurmukhametov** [comment]
Fixed since at least 1.18.0. Test is added by #2636
